lib/filewatch/bootstrap.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
+require "rbconfig"
+require "pathname"
+# require "logstash/environment"
+
+## Common setup
+#  all the required constants and files
+#  defined in one place
+module FileWatch
+  # the number of bytes read from a file during the read phase
+  FILE_READ_SIZE = 32768
+  # the largest fixnum in ruby
+  # this is used in the read loop e.g.
+  # @opts[:file_chunk_count].times do
+  # where file_chunk_count defaults to this constant
+  FIXNUM_MAX = (2**(0.size * 8 - 2) - 1)
+
+  require_relative "helper"
+
+  if LogStash::Environment.windows?
+    require "winhelper"
+  end
+
+  module WindowsInode
+    def prepare_inode(path, stat)
+      fileId = Winhelper.GetWindowsUniqueFileIdentifier(path)
+      [fileId, 0, 0] # dev_* doesn't make sense on Windows
+    end
+  end
+
+  module UnixInode
+    def prepare_inode(path, stat)
+      [stat.ino.to_s, stat.dev_major, stat.dev_minor]
+    end
+  end
+
+  require "java"
+  require_relative "../../lib/jars/filewatch-1.0.0.jar"
+  require "jruby_file_watch"
+
+  if LogStash::Environment.windows?
+    FileOpener = FileExt
+  else
+    FileOpener = ::File
+  end
```

Comment:
maybe this could be brought up and merged with the other `if LogStash::Environment.windows?` check?